### DISCUSSION TITLE
Make use of new indexeddb StorageBucket API

### DISCUSSION THREAD

**Discussion by nickchomey** [dexie/Dexie.js#2014] [Ideas]
The Storage Bucket API ([Background blog](https://developer.chrome.com/docs/web-platform/storage-buckets)) in Chromium browsers allows for better control of persistence and eviction. It is now also more performant, with a sort of [multi-threaded implementation](https://developer.chrome.com/blog/maximum-idb-performance-with-storage-buckets). 

It would be beneficial if Dexie could implement its use in browsers that support it

**Comment by dfahlander** [comment]
It should already be possible:

```ts
const bucket = await navigator.storageBuckets.open('my-bucket');
const db = new Dexie('my-db', { indexedDB: bucket.indexedDB });
```
However, it could probably also be worth it to support it more user-friendly in Dexie. For example:

```ts
const db = new Dexie('my-db', {
  bucket: 'my-bucket',
  persisted: true // passed to storageBuckets.open()
});
```

Question would be how to behave on browsers without the storagebuckets api? Probably to ignore the bucket and persisted arguments, or maybe default persisted to try StorageManager.persist(). But then what happens when the same user upgrades their browser and it gets this API? Probably user will suddenly get an empty DB now that the bucket is created.

Comments?

**Reply by nickchomey** [reply]
Thanks for the reply! Yes, I was suggesting a more user-friendly way.

The blog post I linked to shows some code that checks for support and uses it if available. I'm having trouble making a code block from my phone, so here's a link to it in the post https://developer.chrome.com/blog/maximum-idb-performance-with-storage-buckets#same-site




`
